### ISSUE TITLE
Don't change CID on peer CID change

### ISSUE THREAD

**Issue by ekr** [quicwg/base-drafts#2145]
```Endpoints that use connection IDs with length greater than zero could have their
activity correlated if their peers keep using the same destination connection ID
after migration. Endpoints that receive packets with a previously unused
Destination Connection ID SHOULD change to sending packets with a connection ID
that has not been used on any other network path.  The goal here is to ensure
that packets sent on different paths cannot be correlated. To fulfill this
privacy requirement, endpoints that initiate migration and use connection IDs
with length greater than zero SHOULD provide their peers with new connection IDs
before migration.

Caution:

: If both endpoints change connection ID in response to seeing a change in
  connection ID from their peer, then this can trigger an infinite sequence of
  changes.
```

I don't remember this being what we agreed, and it's not necessary. You only need to change when you see a new path, not after a new CID. That's how we got rid of counting to infinity.

**Comment by mikkelfj** [comment]
I'm not sure it is always well-defined when you are on a new path and you risk accidental linkage. I believe this was the thinking behind this.

**Comment by MikeBishop** [comment]
IIRC, @ekr is correct; we agreed to do this on peer address change.

- If you don't know you've been rebound and you reuse a CID, you've already linked yourself; the peer changing CID in response is pointless.
- If you either anticipate a rebinding or proactively change ports, and change both CID and address simultaneously, then it doesn't matter which of those triggers the peer to switch.
- If you switch CID without changing address, the address links you with fairly high probability anyway, so it doesn't really matter if the peer switches.

**Comment by martinthomson** [comment]
The recommendation exists to avoid linkability across a non-migrating change in connection ID.

**Comment by ekr** [comment]
I don't think that that's particularly useful, and it seems to create a bunch of new problems.

**Comment by janaiyengar** [comment]
My recollection is what @MikeBishop says.  I remember discussing that changing CID on both sides allows a flow to be broken up into flowlets, but if the address is the same on both sides, that's a pretty clear signal anyways. I am fine with requiring a change only if the peer's CID and address change.

**Comment by mikkelfj** [comment]
>  but if the address is the same on both sides, that's a pretty clear signal anyways. I am fine with requiring a change only if the peer's CID and address change.

That is not sufficient because the client could be migrating from domestic wifi to mobile, to train wifi, while the server always sees the IP address of the load balancer.

**Comment by ekr** [comment]
In this case the LB and the server should be regarded as the same thing,
and it's the LB's job to convey that somehow.

**Comment by kazuho** [comment]
FWIW, it helps when you have multiple connections coalesced onto a 5-tuple.

An endpoints rotates the sending CIDs of all the connections at once, and that triggers the peer to rotate all of it's sending CIDs at once. Then, it becomes very hard for an observer to track the individual connections.

**Comment by mnot** [comment]
Tokyo:

> MT has a PR that does just this.

**Comment by martinthomson** [comment]
That being #2414 (I'm closing #2413).

**Comment by ianswett** [comment]
This was closed by #2386
